paddle/fluid/operators/interpolate_op_npu.cc

```diff
@@ -0,0 +1,211 @@
+/* Copyright (c) 2021 PaddlePaddle Authors. All Rights Reserve.
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+   http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#include "paddle/fluid/operators/interpolate_op.h"
+#include <string>
+#include <vector>
+#include "paddle/fluid/operators/npu_op_runner.h"
+
+namespace paddle {
+namespace operators {
+using Tensor = framework::Tensor;
+using DataLayout = framework::DataLayout;
+
+inline static void CheckArgument(const framework::ExecutionContext& ctx) {
+  const std::string interp_method = ctx.Attr<std::string>("interp_method");
+  bool align_corners = ctx.Attr<bool>("align_corners");
+  PADDLE_ENFORCE_EQ(
+      align_corners, false,
+      platform::errors::InvalidArgument(
+          "NPU Interpolate Kernel has diff when align_corners is true."));
+  PADDLE_ENFORCE_EQ(
+      interp_method, "nearest",
+      platform::errors::InvalidArgument(
+          "NPU Interpolate Kernel only support nearest interpolotion."));
+}
+
+inline static void ExtractNCHW(const framework::DDim& dims,
+                               const DataLayout& data_layout, int32_t* n,
+                               int32_t* c, int32_t* h, int32_t* w) {
+  *n = dims[0];
+  if (data_layout == DataLayout::kNCHW) {
+    *c = dims[1];
+    *h = dims[2];
+    *w = dims[3];
```

Comment:
It seems this op only supports tensor with Rank 4?

Reply:
Yes, this OP only supports the nearest interpolation for 2D images, with Rank 4  for the input tensor (n, c, h, w). I got this information from the CPU kernel in "interpolate_op.h", where the "NearestNeighborInterpolate" function is only called for 2D interpolation.